server/server_test.go

```diff
@@ -247,6 +247,17 @@ func TestClientAdvertiseConnectURL(t *testing.T) {
 		t.Fatalf("Expected port to be set to 7777")
 	}
 	s.Shutdown()
+
+	opts = DefaultOptions()
+	opts.Cluster.Port = 0
+	opts.ClientAdvertise = "nats.example.com:7777"
+	s = New(opts)
+	if s.info.Host != "nats.example.com" && s.info.Port != 7777 {
```

Comment:
As described in prior review, I would put a `defer s.Shutdown()` here (more of our standard coding practice). The t.Fatalf() would cause defer to be executed, which without it, server would not be shutdown. This would cause other tests to fail. I already said before that this is ok since we should not have any failure, but still, good practice to do defer here.